lib/calc-text-width.js

```diff
@@ -5,16 +5,19 @@ const SCALE = 10 // Prevent results like 60.599999999999994
 const calcWidth = (charWidthTable) => {
   const widthTable = charWidthTable.map(w => Math.round(w * SCALE))
   widthTable[64] = widthTable[64] + 6 // Slightly increase width of "@" by 0.6px
+  const fallbackWidth = widthTable[64] // Width as "@" for overflows
 
   return (text, astral) => {
     if (astral) text = [...text]
 
     let total = 0
-    let code = 0
+    let width = 0
     let i = text.length
+    let char = ''
     while (i--) {
-      code = text[i].charCodeAt()
-      total += widthTable[code < 127 ? code : 64] // Width as "@" for overflows
+      char = text[i]
+      width = widthTable[char.charCodeAt()]
+      total += width === undefined ? (widthTable[char.normalize('NFD').charCodeAt()] || fallbackWidth) : width
```

Comment:
This line can be shorter (make [codacy](https://app.codacy.com/app/amio/badgen/pullRequest?prid=2085217) happy):
```js
      total += widthTable[char.charCodeAt()] ||
        widthTable[char.normalize('NFD').charCodeAt()] ||
        fallbackWidth
```

Reply:
That was my first implementation, but a width of `0` will make it fall through even though it's not expected. (Very marginal case I admit)

Reply:
Those `[0,0,0...` in width table are control characters, I think we won't meet them in real world use cases, I put them in widthTable mainly for perf reason (avoid a sparse array)

Reply:
We can deal with this case when someone report it :P